src/Microsoft.TestPlatform.PlatformAbstractions/net451/System/PlatformThread.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.PlatformAbstractions
+{
+    using System;
+    using System.Threading;
+
+    using Microsoft.VisualStudio.TestPlatform.PlatformAbstractions.Interfaces;
+
+    public class PlatformThread : IThread
+    {
+        /// <inheritdoc/>
+        public void Run(Action action, PlatformApartmentState apartmentState, bool waitForCompletion)
+        {
+            if (action == null)
+            {
+                return;
+            }
+
+            Exception exThrown = null;
+            var thread = new System.Threading.Thread(() =>
+            {
+                try
+                {
+                    action();
+                }
+                catch (Exception e)
+                {
+                    exThrown = e;
+                }
+            });
+
+            ApartmentState state = Enum.TryParse(apartmentState.ToString(), out state) ? state : ApartmentState.MTA;
```

Comment:
Should we notify user if we are not running test in STA? Because when we are here, we are under the impression that we are going to run test in STA.